crates/ide/src/expand_macro.rs

```diff
@@ -228,6 +229,32 @@ mod tests {
         expect.assert_eq(&actual);
     }
 
+    #[test]
+    fn only_expand_allowed_builtin_macro() {
+        let fail_tests = [r#"
+        //- minicore: asm
+        $0asm!("0x300, x0");
+            "#];
+
+        for test in fail_tests {
+            let (analysis, pos) = fixture::position(test);
+            let expansion = analysis.expand_macro(pos).unwrap();
+            assert!(expansion.is_none());
+        }
+
+        let tests = [(
+            r#"
+            //- minicore: concat
+            $0concat!("test", 10, 'b', true);"#,
+            expect![[r#"
+                concat!
+                "test10btrue""#]],
+        )];
+        for (test, expect) in tests {
+            check(test, expect);
+        }
+    }
+
```

Comment:
Let's just write these out as tests like the rest, that is one test for the failure case, one for success. This here makes it harder to read what is being tested

Reply:
Noted, I'll update